File: strings/344.ReverseString.cpp
//My Solution
//o(n) time and O(1) space
class Solution {
public:
    void reverseString(vector<char>& s) {
        int i=0;
        int j=s.size()-1;
        char temp=' ';
        while(i<j){
            temp=s[i];
            s[i]=s[j];
            s[j]=temp;
            ++i;
            --j;
        }
    }
};

class Solution {
public:
    void reverseString(vector<char>& s) {
        reverse(s.begin(),s.end());
    }
};

File: recursion/344.reverse_string.py

class Solution:
    def reverseString(self, s: List[str]) -> None:
        count = 0
        if(len(s) == 0):
            return
        self.helper(count, s)

    def helper(self, count, s: List[str]) -> List[str]:
        """
        Do not return anything, modify s in-place instead.
        """
        if(count == len(s)//2):
            return s
        temp = s[count]
        s[count] = s[len(s)-count-1]
        s[len(s)-count-1] = temp
        count += 1
        self.helper(count, s)
